From 8ad5ba618ea882692c60e876ee28f8f628661864 Mon Sep 17 00:00:00 2001
From: Steve Arnold <nerdboy@gentoo.org>
Date: Wed, 14 Aug 2019 14:41:36 -0700
Subject: [PATCH 30/38] rtl8723cs: add use of ktime_get_boottime_ts64 for
 kernel versions >= 4.18

* fixes rtl8723cs driver build in newer kernel sources

Signed-off-by: Steve Arnold <nerdboy@gentoo.org>
---
 .../realtek/rtl8723cs/os_dep/linux/ioctl_cfg80211.c         | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtl8723cs/os_dep/linux/ioctl_cfg80211.c b/drivers/net/wireless/realtek/rtl8723cs/os_dep/linux/ioctl_cfg80211.c
index 48a52201d4e6..e863de640b28 100644
--- a/drivers/net/wireless/realtek/rtl8723cs/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/realtek/rtl8723cs/os_dep/linux/ioctl_cfg80211.c
@@ -347,7 +347,11 @@ rtw_cfg80211_default_mgmt_stypes[NUM_NL80211_IFTYPES] = {
 
 static u64 rtw_get_systime_us(void)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0))
+	struct timespec64 ts;
+	ktime_get_boottime_ts64(&ts);
+	return ((u64)ts.tv_sec*1000000) + ts.tv_nsec / 1000;
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
 	struct timespec ts;
 	get_monotonic_boottime(&ts);
 	return ((u64)ts.tv_sec*1000000) + ts.tv_nsec / 1000;
-- 
2.20.1

